src/prng.rs

```diff
@@ -7,51 +7,109 @@
 
 use super::field::{FieldElement, FieldError};
 use aes::{
-    cipher::{generic_array::GenericArray, FromBlockCipher, NewBlockCipher, StreamCipher},
+    cipher::{
+        generic_array::GenericArray, FromBlockCipher, NewBlockCipher,
+        StreamCipher as AesStreamCipher,
+    },
     Aes128, Aes128Ctr,
 };
 use getrandom::getrandom;
 
 use std::marker::PhantomData;
 
-const BLOCK_SIZE: usize = 16;
-const DEFAULT_BUFFER_SIZE_IN_ELEMENTS: usize = 128;
-const MAXIMUM_BUFFER_SIZE_IN_ELEMENTS: usize = 4096;
+const BUFFER_SIZE_IN_ELEMENTS: usize = 128;
 
-pub(crate) const SEED_LENGTH: usize = 2 * BLOCK_SIZE;
-
-pub(crate) fn secret_share<F: FieldElement>(share1: &mut [F]) -> Result<Vec<u8>, getrandom::Error> {
-    let mut seed = vec![0; SEED_LENGTH];
-    getrandom(seed.as_mut_slice())?;
+pub(crate) fn secret_share<F: FieldElement>(share1: &mut [F]) -> Result<Vec<u8>, PrngError> {
```

Comment:
`split` is really only used for tests, I think if we kill anything we should kill that :) The only reason `split` is `pub` is because it's used for examples in the `pcp` module.

That said, both `secret_share` and `extract_share_from_seed` are remnants from the original v2 code. I'd be happy to remove them, but can it wait for a later pr?

Reply:
Yes, like I said, I think we should avoid any changes that are risky Priov2 until we have sufficient test coverage in place to ensure we don't regress. Can you please file an issue so we don't forget to clean this up?

Reply:
you got it: https://github.com/abetterinternet/libprio-rs/issues/78

Reply:
We have at least one test vector for `extract_share_from_seed()` in prng, but I agree that regression testing would be useful. It might also be worth expanding test vectors for v2.